modules/repository/generate.go

```diff
@@ -0,0 +1,229 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repository
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+)
+
+func generateExpansion(src string, templateRepo, generateRepo *models.Repository) string {
+	return os.Expand(src, func(key string) string {
+		switch key {
+		case "REPO_NAME":
+			return generateRepo.Name
+		case "TEMPLATE_NAME":
+			return templateRepo.Name
+		case "REPO_DESCRIPTION":
+			return generateRepo.Description
+		case "TEMPLATE_DESCRIPTION":
+			return templateRepo.Description
+		case "REPO_OWNER":
+			return generateRepo.MustOwnerName()
+		case "TEMPLATE_OWNER":
+			return templateRepo.MustOwnerName()
+		case "REPO_LINK":
+			return generateRepo.Link()
+		case "TEMPLATE_LINK":
+			return templateRepo.Link()
+		case "REPO_HTTPS_URL":
+			return generateRepo.CloneLink(models.DefaultDBContext()).HTTPS
```

Comment:
Can this create problems with SQLite? I mean, there's already a transaction going on on the caller.